src/main/java/org/opentripplanner/routing/algorithm/raptor/router/RaptorRouter.java

```diff
@@ -0,0 +1,127 @@
+import org.opentripplanner.routing.algorithm.raptor.router.street.AccessEgressRouter;
+import org.opentripplanner.routing.algorithm.raptor.router.street.TransferToAccessEgressLegMapper;
+import org.opentripplanner.routing.algorithm.raptor.transit.Transfer;
+import org.opentripplanner.routing.algorithm.raptor.transit.TransitLayer;
+import org.opentripplanner.routing.algorithm.raptor.transit.TripSchedule;
+import org.opentripplanner.routing.algorithm.raptor.transit.request.RaptorRoutingRequestTransitData;
+import org.opentripplanner.routing.core.RoutingRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Does a complete transit search, including access and egress legs.
+ */
+public class RaptorRouter {
+    private final TransitDataProvider<TripSchedule> otpRRDataProvider;
+    private final TransitLayer transitLayer;
+    private static final Logger LOG = LoggerFactory.getLogger(RaptorRouter.class);
+
+    private static final RangeRaptorService<TripSchedule> rangeRaptorService = new RangeRaptorService<>(
+            // TODO - Load turning parameters from config file
+            new TuningParameters() {}
+    );
+
+    //TODO Naming
+    public RaptorRouter(RoutingRequest request, TransitLayer transitLayer) {
+        double startTime = System.currentTimeMillis();
+        this.otpRRDataProvider = new RaptorRoutingRequestTransitData(
+                transitLayer, startDate(request),2, request.modes, request.walkSpeed
+        );
+        LOG.info("Filtering tripPatterns took {} ms", System.currentTimeMillis() - startTime);
+        this.transitLayer = transitLayer;
+    }
+
+    public TripPlan route(RoutingRequest request) {
+
+        /* Prepare access/egress transfers */
+
+        double startTimeAccessEgress = System.currentTimeMillis();
+
+        Map<Stop, Transfer> accessTransfers =
+            AccessEgressRouter.streetSearch(request, false, Integer.MAX_VALUE);
+        Map<Stop, Transfer> egressTransfers =
+            AccessEgressRouter.streetSearch(request, true, Integer.MAX_VALUE);
+
+        TransferToAccessEgressLegMapper accessEgressLegMapper = new TransferToAccessEgressLegMapper(transitLayer);
+
+        Collection<TransferLeg> accessTimes = accessEgressLegMapper.map(accessTransfers, request.walkSpeed);
+        Collection<TransferLeg> egressTimes = accessEgressLegMapper.map(egressTransfers, request.walkSpeed);
+
+        LOG.info("Access/egress routing took {} ms", System.currentTimeMillis() - startTimeAccessEgress);
+
+        /* Prepare transit search */
+
+        double startTimeRouting = System.currentTimeMillis();
+
+        // TODO Time zones
+        int departureTime = Instant.ofEpochMilli(request.dateTime * 1000).atZone(ZoneId.systemDefault()).toLocalTime().toSecondOfDay();
```

Comment:
Should this be `request.getRoutingContext().graph.getTimeZone().toZoneId()` instead of `ZoneId.systemDefault()`? (And the same in `startDate()` lower down).

Reply:
This will depend entirely on time zone semantics which have never been defined clearly. In some cases where it seems like there's a clear answer we discover that some users actually want the other behavior. The solution we seem to be leaning toward long term is a very strict API that does not do any guesswork, which might require a fully-specified search time as a GMT ISO-datetime or with an attached timezone.

Reply:
The timezone handling here is for use within the RAPTOR algorithm with `TripSchedules` to calculate the `seconds since midnight` and the `start of (second) time`, so it needs to be consistent throughout. I've started on patch to handle times consistently as `seconds since GTFS midnight (noon-12)` on the day of planning in the graph timezone. Which is what `TripTime`s already use which are then used to create `TripSchedules`.

Reply:
It looks like this was resolved in 0b27c42, thanks for the contribution.